src/decode.rs

```diff
@@ -4902,19 +4898,19 @@ pub(crate) unsafe fn rav1d_decode_frame(f: &mut Rav1dFrameContext) -> Rav1dResul
     if res.is_ok() {
         if (*f.c).n_tc > 1 {
             res = rav1d_task_create_tile_sbrow(f, 0, 1);
-            pthread_mutex_lock(&mut (*f.task_thread.ttd).lock);
-            pthread_cond_signal(&mut (*f.task_thread.ttd).cond);
+            let mut thread_lock = (*f.task_thread.ttd).delayed_fg.lock().unwrap();
```

Comment:
I don't think we should exactly do that. I have made all the names consistent now (either `task_thread_lock` or `delayed_fg`), but I don't want everything to be `delayed_fg`. There's actually two uses of this lock in the original C code that we can't really do in Rust (and might have chosen to use two different mutexes for, but that would be a very invasive and risky change now). There is access to `delayed_fg`, which needs to be locked for mutation. But there is also usage, such as here, where we are using the lock as a generic task thread lock for synchronization and not actually accessing any shared mutable data. Where we are doing this, and not accessing the field contents, I've named the variable `task_thread_lock` to indicate its use, rather than `delayed_fg`, which it has nothing to do with.

Reply:
I just noticed there is actually a `Rav1dFrameContext_task_thread.lock` field, so maybe these should be called `task_thread_data_lock` instead, but we definitely shouldn't just call them all `delayed_fg` because the locking pattern sometimes has nothing to do with that data.

Reply:
Okay, that makes sense then.  I had assumed all of its uses were to access `delayed_fg`.  Your naming is much better then, since it's clear when `task_thread_lock` is just used for synchronization of external data.

> There's actually two uses of this lock in the original C code that we can't really do in Rust (and might have chosen to use two different mutexes for, but that would be a very invasive and risky change now).

This is referring to moving the other data protected by `task_thread_lock` actually inside the `Mutex`, right?  That's unfortunate but okay for now.

Reply:
Could you add some of this explanation as docs for the `delayed_fg: Mutex<_>` field though?